### DISCUSSION TITLE
Right-angled substructures when filling missing data with Half Quadratic filter

### DISCUSSION THREAD

**Discussion by matheusGariglio** [mtex-toolbox/mtex#1056] [Ask Anything] [answered]
Hi,

In some of my ebsd maps, I have around 50% of non-indexed ebsd data. Thus, I try to fill them with the script below...
```
F = halfQuadraticFilter;

% interpolate the missing data
ebsdSub_filled = smooth(ebsdSub,F,'fill',grainsSub);
ebsdSub_filled = ebsdSub_filled('indexed');
```
...in order to calculate the GND density. 

I usually get substructures that are right-angled (like squares), as you can see in the image below:
![image](https://user-images.githubusercontent.com/77332633/119632907-84ffa500-be11-11eb-9733-ff0a3a7dc036.png)
I would like to know if this is something related to the denoising that tends to form this right-angled substructures.

Please find the raw data and the filtered data in the following link : [link to ctfs](https://cloud.mines-paristech.fr/index.php/s/f8RutixDyo5QXRD)

Cheers,

Matheus

**Comment by kilir** [comment, marked as answer]
Hi,
the rectangular structures for basically in the areas where there is no data.
The gaps in your data are quite large and the GND is based on the local orientation gradient. I would assume the values of GND you will get with your approach will be to a large fraction a function of your filling/smoothing function and be rather unrelated to the material.

Cheers,
Rüdiger

**Reply by matheusGariglio** [reply]
Hi Rüdiger,

Thanks for your answer! Yes, I totally agree, what I've been doing is filling just not indexed areas that are below a certain threshold, to avoid this rectangular structures.

Cheers,
Matheus
